### DISCUSSION TITLE
Issues with Listbox with an image

### DISCUSSION THREAD

**Discussion by NoRuBal** [flamendless/Slab#158] [Q&A] [answered]
Hello. I have two issues on listbox item with an image. I have a code that draws a background image and window with listbox. Listbox contains 10 items with a image and text, which placed in same line. Please check [Listbox_Bug.zip](https://github.com/flamendless/Slab/files/12141004/Listbox_Bug.zip):

**First issue:** Background image turns into dark screen. It does not happen when user placed mouse pointer on scrollbar.
When user placed mouse pointer on the scrollbar:
![problem1](https://github.com/flamendless/Slab/assets/3197094/cb19a0b1-6b00-4635-86a9-5dc262885bf2)
When user didn't placed mouse pointer on elsewhere:
![problem2](https://github.com/flamendless/Slab/assets/3197094/c16315e1-c7c3-4098-945d-712bcdaf495c)

**Second issue:** User can't select last item in the listbox.
![20230724_104800](https://github.com/flamendless/Slab/assets/3197094/f26ce7d9-4e8c-4c84-8e83-abd9178b11da)

If anyone can check those issue it will be very appreciated. I slightly modified the code in SlabTest.lua, so maybe I'm doing something wrong. Thank you for sparing your time and checking my issue. 🙏

**Comment by NoRuBal** [comment]
Update: Just solved the first issue by calling love.graphics.setColor(1, 1, 1, 1) manually.

For second issue, I tried exact same code from the documentation([https://github.com/flamendless/Slab/wiki/ListBox](url)), but it works as same behavior(user can't select last item.):

![20230724_123632](https://github.com/flamendless/Slab/assets/3197094/1b304155-ff96-4d27-9b97-23b52af0c7fd)

```
Slab = require "slab"
local Selected = nil

function love.load()
	Slab.Initialize(args)
end

function love.draw()
	Slab.Draw()
end

function love.update(dt)
	Slab.Update(dt)

	Slab.BeginWindow('ListBox', {Title = "List Box", AutoSizeWindow = false})

	Slab.BeginListBox('ListBoxExample')
	for I = 1, 10, 1 do
		Slab.BeginListBoxItem('ListBoxExample_Item_' .. I, {Selected = Selected == I})
		Slab.Text("Item " .. I)

		if Slab.IsListBoxItemClicked() then
			Selected = I
		end

		Slab.EndListBoxItem()
	end
	Slab.EndListBox()

	Slab.EndWindow()
end
```

I also attach my .love file: [Listbox.zip](https://github.com/flamendless/Slab/files/12141493/Listbox.zip)

**Edit:** The issue is only reproduced on master branch. Using latest release([https://github.com/flamendless/Slab/releases/tag/v0.9.0](url)) of Slab issue dosen't reproduced.

**Comment by flamendless** [comment, marked as answer]
@NoRuBal This is definitely a bug, please file an [issue](https://github.com/flamendless/Slab/issues/new) so we can track it better :) the fix should be easy

**Reply by NoRuBal** [reply]
Thank you for checking it! I'll write an issue.
